src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement/Services/NationalAchievementRates/NationalAchievementRatesPageParser.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using AngleSharp;
+using AngleSharp.Dom;
+
+namespace SFA.DAS.RoatpCourseManagement.Services.NationalAchievementRates
+{
+    public class NationalAchievementRatesPageParser : INationalAchievementRatesPageParser
+    {
+        private static string NationalAchievementRatesPageUrl => "https://www.gov.uk/government/statistics/national-achievement-rates-tables-{0}-to-{1}";
+
+        public async Task<string> GetCurrentDownloadFilePath()
+        {
+            var yearTo = DateTime.Today.Year;
+            var yearFrom = DateTime.Today.AddYears(-1).Year;
+            var config = AngleSharp.Configuration.Default.WithDefaultLoader();
+            var context = BrowsingContext.New(config);
+            IDocument document = null;
+            var pageFound = false;
+            while (!pageFound)
+            {
+                document = await context.OpenAsync(string.Format(NationalAchievementRatesPageUrl, yearFrom, yearTo));
+                if (document.StatusCode != HttpStatusCode.NotFound)
+                {
+                    pageFound = true;
+                }
+                else
+                {
+                    yearTo--;
+                    yearFrom--;
+                }
+            }
```

Comment:
If the URL changes for NAR, this will potentially go in to endless loop. I suggest we put a check here. If the loop has done 10 iterations, this should log an error and throw exception.  Following code can be added to the end of the loop.
```
if (yearTo < DateTime.Today.Year.AddYears(-10)) 
{
  log.Error
  throw new InvalidOperationException
}